src/nft/pages/asset/Asset.tsx

```diff
@@ -1,37 +1,289 @@
+  })
 
-  let asset = {} as GenieAsset
-  let collection = {} as CollectionInfoForAsset
+  useEffect(() => {
+    if (asset.creator) setCreatorAddress(asset.creator.address)
+    if (asset.owner) setOwnerAddress(asset.owner)
+  }, [asset])
 
-  if (data) {
-    asset = data[0] || {}
-    collection = data[1] || {}
-  }
+  const { rarityProvider } = useMemo(
+    () =>
+      asset.rarity
+        ? {
+            rarityProvider: asset.rarity.providers.find(
+              ({ provider: _provider }) => _provider === asset.rarity?.primaryProvider
+            ),
+            rarityLogo: rarityProviderLogo[asset.rarity.primaryProvider] || '',
+          }
+        : {},
+    [asset.rarity]
+  )
+
+  const assetMediaType = useMemo(() => {
+    if (isAudio(asset.animationUrl)) {
+      return 'audio'
+    } else if (isVideo(asset.animationUrl)) {
+      return 'video'
+    }
+    return 'image'
+  }, [asset])
 
   return (
-    <div>
-      {' '}
-      <Details
-        contractAddress={contractAddress}
-        tokenId={tokenId}
-        tokenType={asset.tokenType}
-        blockchain="Ethereum"
-        metadataUrl={asset.externalLink}
-        totalSupply={collection.totalSupply}
-      />
-    </div>
+    <AnimatedBox
+      style={{
+        // @ts-ignore
+        width: gridWidthOffset.interpolate((x) => `calc(100% - ${x}px)`),
+      }}
+      className={styles.container}
+    >
+      <div className={styles.columns}>
+        <Column className={styles.column}>
+          {assetMediaType === 'image' ? (
+            <img
+              className={styles.image}
+              src={asset.imageUrl}
+              alt={asset.name || collection.collectionName}
+              style={{ ['--shadow' as string]: `rgba(${dominantColor.join(', ')}, 0.5)` }}
+            />
+          ) : (
+            <AssetView asset={asset} mediaType={assetMediaType} dominantColor={dominantColor} />
+          )}
+        </Column>
+        <Column className={clsx(styles.column, styles.columnRight)} width="full">
+          <Column>
+            <Row marginBottom="8" alignItems="center" justifyContent={rarityProvider ? 'space-between' : 'flex-end'}>
+              <Row gap="12">
+                <Center
+                  as="button"
+                  padding="0"
+                  border="none"
+                  background="transparent"
+                  onClick={async () => {
+                    await navigator.clipboard.writeText(window.location.hostname + pathname)
+                  }}
+                >
+                  <ShareIcon />
+                </Center>
+
+                <Center
+                  as="button"
+                  border="none"
+                  width="32"
+                  height="32"
+                  padding="0"
+                  background="transparent"
+                  cursor="pointer"
+                  onClick={() => {
+                    if (!parsed.origin || parsed.origin === 'collection')
```

Comment:
nit: this is hard to read without using new lines for the if block. can you add the brackets for these branches?